src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3LinuxDriver.cs

```diff
@@ -350,6 +356,96 @@ private void SetInputPullModePi4(int pinNumber, PinMode mode)
             }
         }
 
+        /// <inheritdoc />
+        protected internal override void SetExtendedPinMode(int pinNumber, ExtendedPinMode altMode)
+        {
+            Initialize();
+            ValidatePinNumber(pinNumber);
+
+            RaspberryPiExtendedPinMode? raspiMode = altMode as RaspberryPiExtendedPinMode;
+            if (ReferenceEquals(raspiMode, null))
+            {
+                throw new ArgumentException(nameof(altMode), $"Invalid extended mode: {raspiMode}");
+            }
+
+            /*
+             * There are 6 registers (4-byte ints) that control the mode for all pins. Each
+             * register controls the mode for 10 pins. Each pin uses 3 bits in the register
+             * containing the mode.
+             */
+
+            // Define the shift to get the right 3 bits in the register
+            int shift = (pinNumber % 10) * 3;
+            // Gets a pointer to the register that holds the mode for the pin
+            uint* registerPointer = &_registerViewPointer->GPFSEL[pinNumber / 10];
+            uint register = *registerPointer;
+            // Clear the 3 bits to 0 for the pin Number.
+            register &= ~(0b111U << shift);
+            // Set the 3 bits to the desired mode for that pin.
+            uint modeBits = 0; // Default: Gpio input
+
+            if (raspiMode == RaspberryPi3Driver.GpioMode)
+            {
+                // When setting back to Gpio, set to input, as this is the default
+                modeBits = 0;
+            }
+            else if (raspiMode.ModeValue < 8)
+            {
+                modeBits = raspiMode.ModeValue;
+            }
+            else
+            {
+                throw new InvalidOperationException($"Unknown Alternate pin mode value: {raspiMode.ModeValue}");
+            }
+
+            register |= (modeBits) << shift;
+            *registerPointer = register;
+        }
+
+        /// <inheritdoc />
+        protected internal override ExtendedPinMode GetExtendedPinMode(int pinNumber)
+        {
+            Initialize();
+            ValidatePinNumber(pinNumber);
+            /*
+             * There are 6 registers(4-byte ints) that control the mode for all pins. Each
+             * register controls the mode for 10 pins. Each pin uses 3 bits in the register
+             * containing the mode.
+             */
+
+            // Define the shift to get the right 3 bits in the register
+            int shift = (pinNumber % 10) * 3;
+            // Gets a pointer to the register that holds the mode for the pin
+            uint* registerPointer = &_registerViewPointer->GPFSEL[pinNumber / 10];
+            uint register = *registerPointer;
+            // get the three bits of the register
+            register = (register >> shift) & 0b111;
+
+            switch (register)
+            {
+                case 0b000:

```

Comment:
since you have exactly 8 values it's faster to create an array with all entries:
`ExtendedPinMode[] s_pinModes = new[8] { ... };`

and then just use indexer here

Reply:
you can put this array in the RaspberryPiExtendedPinMode and have a method in there which resolves int into the instance

Reply:
Not sure this is a good idea. For one, this would cause a static initializer depenceny between this class and RaspberryPi3Driver, and then it may be confusing, because the order is not sequential.